pydantic/main.py

```diff
@@ -360,11 +360,14 @@ def __init__(__pydantic_self__, **data: Any) -> None:
         values, fields_set, validation_error = validate_model(__pydantic_self__.__class__, data)
         if validation_error:
             raise validation_error
-        object_setattr(__pydantic_self__, '__dict__', values)
+        try:
+            object_setattr(__pydantic_self__, '__dict__', values)
+        except TypeError as e:
+            raise TypeError('Did you return a valid dict in your root validators?') from e
         object_setattr(__pydantic_self__, '__fields_set__', fields_set)
         __pydantic_self__._init_private_attributes()
 
-    @no_type_check
+    @no_type_check  # noqa: C901 (ignore complexity)
```

Comment:
this looks unrelated.

Reply:
It's because of https://github.com/samuelcolvin/pydantic/pull/2198#pullrequestreview-550921099